avae/data.py

```diff
@@ -123,209 +118,209 @@ def load_data(
+
+    Parameters
+    ----------
+    affinity: str | None
+        Path to the affinity matrix if provided .
+    classes: list
+        List of classes to be selected from the data.
+
+    Returns
+    -------
+    affinity: pd.DataFrame or None if no affinity matrix path is provided
+        Affinity matrix.
+    """
+    if affinity_path is not None:
+        # load affinity matrix
+        affinity = pd.read_csv(affinity_path, header=0)
+    else:
+        affinity = None
+
+    if affinity is not None:
+        class_check = np.in1d(classes, affinity.columns)
+        if not np.all(class_check):
+            raise RuntimeError(
+                "Not all classes in the training set are present in the "
+                "affinity matrix. Missing classes: {}".format(
+                    np.asarray(classes)[~class_check]
                 )
+            )
+        if settings.VIS_AFF:
+            plot_affinity_matrix(
+                lookup=affinity,
+                all_classes=affinity.columns.tolist(),
+                selected_classes=classes,
+            )
+
+        # subset affinity matrix with only the relevant classes
+        index = [affinity.columns.get_loc(f"{columns}") for columns in classes]
+        sub_affinity = affinity.iloc[index, index]
+
+        return sub_affinity
 
-            # subset affinity matrix with only the relevant classes
-            index = [
-                self.amatrix.columns.get_loc(f"{columns}")
-                for columns in self.final_classes
-            ]
-            self.amatrix = self.amatrix.iloc[index, index]
+    else:
+        return None
 
-        self.paths = [
-            p
-            for p in self.paths
-            for c in self.final_classes
-            if c in p.split("_")[0]
-        ]
 
-        self.paths = self.paths[:lim]
+class AffinityDiskDataset(DiskDataset):
 
-    def __len__(self):
-        return len(self.paths)
+    """Modified version of the caked DiskDataset to include the affinity matrix and data metadata that is needed for the
+    affinity pipeline"""
 
-    def dim(self):
-        return len(np.array(self.read(self.paths[0])).shape)
+    def __init__(
+        self,
+        dataset: DiskDataset,
+        classes: list,
+        affinity: pd.DataFrame | None = None,
+    ):
+        super().__init__(
+            paths=dataset.paths,
+            datatype=dataset.datatype,
+            rescale=dataset.rescale,
+            shiftmin=dataset.shiftmin,
+            normalise=dataset.normalise,
+            gaussianblur=dataset.gaussianblur,
+        )
+        self.affinity = affinity
+        self.classes = classes
 
-    def __getitem__(self, item):
-        filename = self.paths[item]
+    def __getitem__(self, index):
 
-        data = np.array(self.read(filename))
-        x = self.voxel_transformation(data)
+        # read data from path
+        data = np.array(self.read(self.paths[index]))
+        x = self.transformation(data)
```

Comment:
what is self.transformation() ?

Reply:
this is the equivalent to voxel_transformation, but as we should be able to deal with very different datasets (1D, 2D and 3D) it was more clear to leave it a `transformation` only.